py/parse_and_notify.py

```diff
@@ -50,29 +68,41 @@ class NewTopic(Topic):
     change_set: Sequence[Sequence[str | int | None]]
     dry_run: bool
     error: bool
+    error_message: str | None
     name: str
 
     def render_table(self) -> str:
         return make_markdown_table(
             headers=["Parameter", "Value"],
             content=[["Topic Name", self.name], *self.change_set],
+            error=self.error,
+            error_message=self.error_message,
         )
 
     @classmethod
     def build(cls, raw_content: Mapping[str, Any]) -> NewTopic:
+        change_set = [["Action (create/update)", "create"]]
+        if raw_content["numPartitions"]:
+            change_set.extend(
+                [["Partition Count", raw_content["numPartitions"]]]
+            )
+        if raw_content["replicationFactor"]:
+            change_set.extend(
+                [["Replication Factor", raw_content["replicationFactor"]]]
+            )
+        change_set += [
+            [str(entry["name"]), str(entry["value"])]
+            for entry in raw_content["configEntries"]
+        ]
+        # if nothing changed, report no changes
+        if len(change_set) == 1:
+            change_set = []
         return NewTopic(
             name=raw_content["topic"],
             dry_run=raw_content["dryRun"],
-            error=False,
-            change_set=[
-                ["Action (create/update)", "create"],
-                ["Partition Count", raw_content["numPartitions"]],
-                ["Replication Factor", raw_content["replicationFactor"]],
-            ]
-            + [
-                [str(entry["name"]), str(entry["value"])]
-                for entry in raw_content["configEntries"]
-            ],
+            error=raw_content["error"],
+            error_message=raw_content["errorMessage"],
+            change_set=change_set,
```

Comment:
Since we can report no changes if an error occurs before any changes happen, we can't assume Partition count / Replication Factor are filled in when we receive the event.